src/Pos/Explorer/Web/Server.hs

```diff
@@ -179,12 +179,14 @@ getLastTxs (fromIntegral -> lim) (fromIntegral -> off) = do
     pure $ [toTxEntry (tiTimestamp txi) (tiTx txi) | txi <- localTxsWithTs <> blockTxsWithTs]
 
 getBlockSummary :: ExplorerMode m => CHash -> m CBlockSummary
-getBlockSummary (fromCHash' -> h) = do
+getBlockSummary cHash = do
+    h <- either (throwM . Internal) pure $ fromCHash cHash
```

Comment:
It's probably better to make it separate helper (like `maybeThrow`). But it's cosmetics